extensions/functions_arithmetic.yaml

```diff
@@ -175,12 +175,20 @@ scalar_functions:
         return: fp64
   -
     name: "divide"
-    description: "Divide one value by another. Partial values are truncated."
+    description: >
+      Divide x by y. In the case of integer division, partial values are truncated.
+      The `on_division_by_zero` option governs behaviour in cases where y is 0 and x is not 0.
+      The `on_domain_error` option governs behaviour for 0/0 and Inf/Inf etc.
     impls:
       - args:
           - name: overflow
             options: [ SILENT, SATURATE, ERROR ]
             required: false
+          - name: on_division_by_zero
+            options: [ NAN, ERROR, LIMIT ]
+            required: false
+          - name: on_domain_error
+            options: [ NAN, ERROR ]
```

Comment:
NaNs and infinities are, thankfully, not a thing for integers (and Substrait is strongly typed, so the function can't decide to return those things anyway). So the integer variants have no choice but to fail for divisions by zero. I think there *are* actually programming languages that define the result of an integer division by zero to zero (which is perfectly fine mathematically, you can define a division by zero to whatever you want), but let's not go that far unless there are also *query engines* that do that.

Reply:
That's useful to know, updated now!